### PR TITLE
Fix #188: Fix Pauli measurement conventions

### LINKED ISSUE #188: [Bug]: Wrong Pauli measurements for axes `YZ` and `XZ`
Axes conventions have been changed in 81e1a652 for measurements during simulation, but Pauli preprocessing was not updated, leading to inconsistencies for Pauli measurements outside the `XY` axis. The following test currently succeeds only on the `XY` axis but not on `YZ` and `XZ`. 

```python
    @pytest.mark.parametrize("plane", Plane)
    @pytest.mark.parametrize("angle", [0., 0.5, 1., 1.5])
    def test_pauli_measurement_single(self, plane: Plane, angle: float, use_rustworkx: bool = True) -> None:
        pattern = Pattern(input_nodes=[0, 1])
        pattern.add(E(nodes=[0, 1]))
        pattern.add(M(node=0, plane=plane, angle=angle))
        pattern_ref = pattern.copy()
        pattern.perform_pauli_measurements(use_rustworkx=use_rustworkx)
        state = pattern.simulate_pattern()
        state_ref = pattern_ref.simulate_pattern(pr_calc=False, rng=IterGenerator([0]))
        assert np.abs(np.dot(state.flatten().conjugate(), state_ref.flatten())) == pytest.approx(1)
```

Axes conventions are defined in `pauli.py`. The current convention is:
| Plane | cos | sin |
|-|-|-|
| XY | X | Y |
| YZ | Z | Y |
| XZ | Z | X |

The former convention was:
| Plane | cos | sin |
|-|-|-|
| XY | X | Y |
| YZ | Y | Z |
| XZ | X | Z |

### PR DESCRIPTION
**Pull request by thierry-martinez**
This commit introduces a new named tuple, `graphix.pauli.Measure`, which contains a pair `(axis, sign)`. It includes a static method, `try_from`, that returns an instance of `Measure` if the given plane and angle represent a valid Pauli measure, or `None` otherwise.

The axis is computed using the `cos` and `sin` definitions from `pauli.py`, addressing the inconsistency bug reported in #188.

Additionally, this commit introduces a new `Sign` class, which enhances the readability of code that manipulates `Plus` and `Minus` values, as opposed to using `bool`.

Note: This PR follows up on #189, focusing on fixing #188 while preserving the equality test for recognizing Pauli measures.

### CODE REVIEW

File: graphix/pauli.py
Comment (shinich1) [inline graphix/pauli.py:364]: perhaps differnet class name is good to make clear it's specifically to work with Pauli measurement?

Comment (thierry-martinez) [reply graphix/pauli.py]: Fixed in e374cd5.

Comment (thierry-martinez) [reply graphix/pauli.py]: This definitely indicates that unification is needed! However, I don’t think this necessarily dictates a specific order for merging:
- Regardless of which PR is merged first, the others can be unified during the rebasing process.
- It seems to me that the primary need for unification is between `MeasurementDescription` (in #178) and `Measurement` (in #191), as both classes are intended for general measurement (and neither of these PRs introduces a class specifically for Pauli measures).
- `pauli.py` seems like an appropriate place for defining `PauliMeasure`.

File: graphix/pattern.py
Comment (shinich1) [inline graphix/pattern.py:1972]:
```suggestion
        pm = PauliMeasure.try_from(cmd.plane, cmd.angle)  # None returned if the measurement is not in Pauli basis
```

File: graphix/pauli.py
Comment (shinich1) [reply graphix/pauli.py]: sorry - could you rename to `PauliMeasurement` to be consistent with other classes mentioned above?

**Comment by shinich1** [comment]
@thierry-martinez thanks! will approve once the above two comments resolved.

File: graphix/pauli.py
Comment (thierry-martinez) [reply graphix/pauli.py]: Fixed in 6388d4.

**Comment by thierry-martinez** [comment]
> @thierry-martinez thanks! will approve once the above two comments resolved.

Done! Thanks @shinich1 .

**Review by shinich1 — APPROVED** [review]

**Comment by EarlMilktea** [comment]
Memo: I confirmed these `Number`-related `mypy` errors are newly introduced after this PR merge.
Let's resolve them in #201 !

```bash
graphix/pauli.py:51: error: Incompatible return value type (got "Sign", expected "SignOrNumber")  [return-value]
graphix/pauli.py:53: error: Incompatible return value type (got "int", expected "SignOrNumber")  [return-value]
graphix/pauli.py:53: error: Unsupported operand types for * ("int" and "Number")  [operator]
graphix/pauli.py:58: error: Incompatible return value type (got "int", expected "Number")  [return-value]
graphix/pauli.py:58: error: Unsupported operand types for * ("int" and "Number")  [operator]
graphix/pauli.py:342: error: No overload variant of "int" matches argument type "Number"  [call-overload]
graphix/pauli.py:356: error: Argument 1 to "is_int" has incompatible type "float"; expected "Number"  [arg-type]
```

**Pull request merged**
